File: backend/src/main/java/com/juansierra/global_invoice_api/service/interfaces/InvoiceService.java
package com.juansierra.global_invoice_api.service.interfaces;

import com.juansierra.global_invoice_api.dto.request.CreateInvoiceRequest;
import com.juansierra.global_invoice_api.dto.response.InvoiceDetailResponse;
import com.juansierra.global_invoice_api.dto.response.InvoiceResponse;
import java.util.List;

public interface InvoiceService {

    InvoiceResponse createInvoice(CreateInvoiceRequest request);

    List<InvoiceResponse> findAll();

    InvoiceDetailResponse findById(Long invoiceId);
}

File: frontend/src/app/core/invoices/invoice.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable, inject } from '@angular/core';
import { Observable } from 'rxjs';
import { CreateInvoiceRequest, InvoiceDetailResponse, InvoiceResponse } from './invoice.models';

@Injectable({ providedIn: 'root' })
export class InvoiceService {
  private readonly http = inject(HttpClient);

  create(request: CreateInvoiceRequest): Observable<InvoiceResponse> {
    return this.http.post<InvoiceResponse>('/api/v1/invoices', request);
  }

  findAll(): Observable<InvoiceResponse[]> {
    return this.http.get<InvoiceResponse[]>('/api/v1/invoices');
  }

  findById(invoiceId: number): Observable<InvoiceDetailResponse> {
    return this.http.get<InvoiceDetailResponse>(`/api/v1/invoices/${invoiceId}`);
  }
}
